### PR TITLE
Added new isUserDev method

### PR DESCRIPTION
**Pull request by pablonortiz**
LINK DE TICKET:
https://janiscommerce.atlassian.net/browse/APPSRN-458

DESCRIPCIÓN DEL REQUERIMIENTO:
Tras identificar que la validación isDev se puede llegar a usar en las tres APPs, se decidió crear un nuevo método en [oauth-native](https://github.com/janis-commerce/oauth-native) para poder reutilizarlo fácilmente a través de las APPs.

DESCRIPCIÓN DE LA SOLUCIÓN:
Se creó el método isUserDev para validar cuándo el usuario es dev a partir del idToken.

NIVEL PRUEBAS * ### (Marcar con X)

[ ] NIVEL CRITICO, CAMBIOS MAYORES

[ ] NIVEL ALTO, CAMBIOS MODERADOS

[ ] NIVEL MEDIO, CAMBIOS MENORES

[X] NIVEL BAJO, CAMBIOS NO CRITICOS

CÓMO SE PUEDE PROBAR?
Probar linkeando a una de las tres apps y verificando que la función devuelva si el usuario es dev o no

SCREENSHOTS:

DATOS EXTRA A TENER EN CUENTA:

CHANGELOG:

---

> [!NOTE]
> Introduces `isUserDev` to detect developer users from the `idToken` (via `jwt-decode`), exports it from the package, and adds tests.
> 
> - **Utils**:
>   - Add `isUserDev` in `src/utils/oauth.js` using `jwt-decode` to read `isDev` from `idToken`.
> - **Exports**:
>   - Re-export `isUserDev` from `src/index.js`.
> - **Tests**:
>   - Add `isUserDev` tests covering true/false, missing token, and null tokens.
>   - Mock `jwt-decode` in tests.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit a174acf5bf3b97f7044eaa3bf7decaf6f358bb89. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

## Summary by CodeRabbit

* **New Features**
  * Added a developer-status detection utility that reads the authenticated user's token to indicate developer privileges, enabling developer-specific behavior in the app.

* **Tests**
  * Added tests covering the new developer detection utility, including successful detection and error/edge-case handling.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/utils/oauth.js [hunk 1]
```diff
@@ -198,3 +199,20 @@ export const isTokenExpired = async () => {
...
+ * @description Asynchronously checks if the user is a developer by retrieving the isDev property from the token.
+ * @async
+ * @returns {Promise<boolean>} - Resolves to true if the user is a developer, false otherwise.
+ */
+export const isUserDev = async () => {
+  try {
+    const {oauthTokens} = await getTokensCache();
+    const {idToken = ''} = oauthTokens || {};
+    const decoded = jwtDecode(idToken);
+    return decoded?.isDev;
```
Comment (GonzaFran) [inline src/utils/oauth.js]: @pablonortiz acá usa la sugerencia de la IA para que siempre se retorne un valor booleano.

```js
return decoded?.isDev ?? false
```

File: src/utils/oauth.js [hunk 2]
```diff
@@ -198,3 +199,20 @@ export const isTokenExpired = async () => {
...
+  } catch (reason) {
+    return Promise.reject(reason);
```
Comment (GonzaFran) [inline src/utils/oauth.js]: @pablonortiz en este caso, como es una validación sencilla, retornaría un false en caso de que entre en el `catch`, para ahorrarnos problemas durante la implementación.

**Review by GonzaFran — CHANGES_REQUESTED** [review]
@pablonortiz dejo 2 comentarios, uno para implementar y el otro para revisar si es conveniente o no.

**Comment by pablonortiz** [comment]
Listos los ajustes @GonzaFran !

**Review by GonzaFran — APPROVED** [review]
